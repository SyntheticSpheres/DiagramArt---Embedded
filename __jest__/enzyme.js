// https://github.com/airbnb/enzyme/issues/1265

const Enzyme = require('enzyme');
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
