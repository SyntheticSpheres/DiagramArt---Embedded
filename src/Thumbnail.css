.da-thumbnail {
  display: block;
  position: relative;
  box-shadow: 0 0 0 1px #ebebeb;
  float: left;
  margin: 5px;
  cursor: pointer;
}

.da-thumbnail.large {
  width: 300px;
  height: 259px;
  max-height: 259px;
  padding-bottom: 34px;
}

.da-thumbnail.small {
  width: 150px;
  height: 129px;
  max-height: 129px;
  padding-bottom: 34px;
}

.da-thumbnail .da-thumbnail-title {
  position: absolute;
  bottom: 0;
  padding: 0 5px;
}

.da-thumbnail .da-thumbnail-title h3 {
  margin: 5px 0;
}

.da-thumbnail.small .da-thumbnail-title h3 {
  font-size: 0.8em;
}


.da-thumbnail:hover .da-thumbnail-title h3 {
    color: #b8e02e;
}

.da-thumbnail .da-thumbnail-picture {
  text-align: center;
}

.da-thumbnail.large .da-thumbnail-picture {
  line-height: 225px;
}

.da-thumbnail.small .da-thumbnail-picture {
  line-height: 112px;
}

.da-thumbnail .da-thumbnail-picture img {
  max-width: 100%;
  height: auto;
  width: auto;
  vertical-align: middle;
}

.da-thumbnail.large .da-thumbnail-picture img {
  max-height: 225px;
}

.da-thumbnail.small .da-thumbnail-picture img {
  max-height: 112px;
}

.da-thumbnail-picture .sk-circle {
  margin: 40px auto;
}
