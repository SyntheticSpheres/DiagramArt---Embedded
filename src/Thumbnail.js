import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImageLoader from './ImageLoader';
import './Thumbnail.css';

class Thumbnail extends Component {
  constructor(props) {
    super(props);

    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.props.onThumbnailClick(this.props.id);
  }

  render() {
    return (
      <div className="da-thumbnail small" onClick={this.onClick}>
        <div className="da-thumbnail-picture">
          <ImageLoader image={this.props.picture} />
        </div>
        <div className="da-thumbnail-title">
          <h3>{this.props.title}</h3>
        </div>
      </div>
    );
  }
}

Thumbnail.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  onThumbnailClick: PropTypes.func.isRequired,
}

export default Thumbnail;
