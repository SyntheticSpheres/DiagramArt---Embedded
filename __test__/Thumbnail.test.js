import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import Thumbnail from '../src/Thumbnail';

describe('Thumbnail component', () => {
  let sandbox;
  let callback;

  beforeEach(function() {
    sandbox = sinon.sandbox.create();
    callback = sandbox.stub();
  });

  afterEach(function() {
    sandbox.restore();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Thumbnail id={1} title="diagram" picture="https://cdn.shopify.com/s/files/1/0255/1589/products/USPTO_Patent_0003302.png" onThumbnailClick={callback} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Thumbnail id={1} title="diagram" picture="https://cdn.shopify.com/s/files/1/0255/1589/products/USPTO_Patent_0003302.png" onThumbnailClick={callback} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('simlate click', () => {
    const tree = shallow(
      <Thumbnail id={1} title="diagram" picture="https://cdn.shopify.com/s/files/1/0255/1589/products/USPTO_Patent_0003302.png" onThumbnailClick={callback} />
    );
    tree.find('.da-thumbnail').simulate('click');
    sinon.assert.calledOnce(callback);
  });
});
