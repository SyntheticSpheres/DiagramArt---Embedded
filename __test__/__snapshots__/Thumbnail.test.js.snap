// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail component renders correctly 1`] = `
<div
  className="da-thumbnail small"
  onClick={[Function]}
>
  <div
    className="da-thumbnail-picture"
  >
    <img
      alt=""
      src="https://cdn.shopify.com/s/files/1/0255/1589/products/USPTO_Patent_0003302.png"
    />
  </div>
  <div
    className="da-thumbnail-title"
  >
    <h3>
      diagram
    </h3>
  </div>
</div>
`;
